### PR TITLE
Added a button that takes the user to a page that will be able to add…

### LINKED ISSUE #22: New Entry Button

### PR DESCRIPTION
**Pull request by caelanot**
… a new entry.

Adds a button to the new entry page from entries.dart

# Description

User will be able to enter a journal entry on this page, which will be shown on the entries page for a specific day.

Fixes #22 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Testing widget existence
- [x] Testing text field inputs match expected results

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by caelanot** [comment]
https://pub.dev/packages/syncfusion_flutter_gauges
https://pub.dev/packages/circular_seek_bar

Looking into these two packages for radial wheel for emotion input.
New entry button is on the entries.dart page (may be moved around later), saving functionality does not exist yet.

File: app/lib/pages/new_entry.dart [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      floatingActionButton: FloatingActionButton(
+        key: const Key("SaveButton"),
+        onPressed: () {
+          Navigator.pop(context);
+        },
+        child: const Text("Save")
+      ),
+    );
+  }
+}
```
Comment (Cooljawty) [inline app/lib/pages/new_entry.dart:91]: These functions should be handeled by the Bar from the entries page.
In fact I think we need to move it from the Entries page to this page.

File: app/lib/pages/entries.dart [hunk 2]
```diff
@@ -34,7 +35,18 @@ class _EntriesPageState extends State<EntriesPage> {
...
+        key: const Key("New Entry Button"),
+        padding: const EdgeInsets.only(bottom: 50),
+        child: FloatingActionButton(
+          onPressed: () {
+            Navigator.push(context, NewEntryPage.route());
+          },
+          child: const Icon(Icons.add),
+        ),
+      )
+    );
```
Comment (Cooljawty) [inline app/lib/pages/entries.dart:49]: This button will be on the navigation bar.
I'd recommend waiting until the #68 is merged then adding the button to the navbar

**Review by Cooljawty — CHANGES_REQUESTED** [review]
In summary:
 - save, plan, tag widget needs to be moved to the new_entry page
 - JournalEntry class needs to be implimented in the save gesture (I've defined the class in #60, so it can wait until thats been merged)
 - The button for new entries should be on the navigaton bar. Unless we've decided to change that from the prototype.

**Comment by caelanot** [comment]
> In summary:
> 
> * save, plan, tag widget needs to be moved to the new_entry page
> * JournalEntry class needs to be implimented in the save gesture (I've defined the class in [29 display card #60](https://github.com/S1robe/Pocket-Therapist/pull/60), so it can wait until thats been merged)
> * The button for new entries should be on the navigaton bar. Unless we've decided to change that from the prototype.

Okay that's what I figured was for the save/plan/ bar to be moved, I wasn't sure yet if the entries.dart page was for displaying entries or adding new ones, which was why I temporarily made the button its own thing and not yet in the bar.
Once #68 and #60 are handled I'll merge them over into this and move the new entry button as well.

**Comment by caelanot** [comment]
@Cooljawty @S1robe Test nessage

File: app/lib/pages/new_entry.dart (hunk 1 above)
Comment (caelanot) [reply app/lib/pages/new_entry.dart]: Moved bar to new_entry page instead

File: app/lib/pages/entries.dart (hunk 2 above)
Comment (caelanot) [reply app/lib/pages/entries.dart]: Temporary button on the page still, will change later

**Comment by Benek6h** [comment]
You need to fix the merge conflicts before we review @caelanot

**Review by ian5924 — APPROVED** [review]
The button test are well done.
Thanks!

**Review by S1robe — CHANGES_REQUESTED** [review]
Code looks good, please run integration tests and commit your lcov.info file after running `flutter test integration_test --branch`

Thank you

**Comment by caelanot** [comment]
> Code looks good, please run integration tests and commit your lcov.info file after running `flutter test integration_test --branch`
> 
> Thank you

Code coverage now updated with integration test results.

**Review by S1robe — COMMENTED** [review]
@caelanot Can you explain the difference between lib/pages/new_entry.dart and lib/pages/entry.dart 

Specifically the pages, it seems like there are two pages for the same thing.
Thank you for adding code coverage.

**Comment by caelanot** [comment]
> @caelanot Can you explain the difference between lib/pages/new_entry.dart and lib/pages/entry.dart
> 
> Specifically the pages, it seems like there are two pages for the same thing. Thank you for adding code coverage.

The new_entry.dart page is what lets the user make a new journal entry, while entry.dart is for displaying an already created journal entry, which should be two separate pages.

**Review by S1robe — APPROVED** [review]
Looks good, thank you for the hard work!

I appreciate the concern for performance by creating two separate widgets rather than combining all the state into one.

Great job.

**Pull request merged**